backend/src/oe/api/op-statistics.service.ts

```diff
@@ -19,36 +21,44 @@ export class OpStatisticService {
   ): Promise<NbdrStatistics | NbdrStatisticsError> {
     try {
       const nbdrStatisticsList: number[] = [];
-      let lastblock = blockHeight.value;
-      const blockNumbers = [] as number[];
-
-      // creating array of last 100 blocks spans
-      for (let i = 0; i < 100; i += 2) {
-        blockNumbers.push(lastblock, lastblock - blockSpan);
-        lastblock = lastblock - (blockSpan + 1);
-      }
-      
       const confirmedBlocks = [] as ConfirmedBlockHeight[];
+
       try {
+        // fetching block span list
+        const blockSpanList = await opEnergyService.$getBlockSpanList(
+          'nbdr',
+          blockHeight.value,
+          blockSpan,
+          NUMBER_OF_BLOCK_SPANS
+        );
         const currentTip = await bitcoinApi.$getBlockHeightTip();
-        blockNumbers.forEach((blockNumber) =>
+
+        blockSpanList.forEach((blockNumber) =>
           confirmedBlocks.push(
-            opBlockHeaderService.verifyConfirmedBlockHeight(blockNumber, {
-              value: currentTip,
-            })
+            opBlockHeaderService.verifyConfirmedBlockHeight(
+              blockNumber.startBlockHeight,
+              {
+                value: currentTip,
+              }
+            ),
+            opBlockHeaderService.verifyConfirmedBlockHeight(
+              blockNumber.endBlockHeight,
+              {
+                value: currentTip,
+              }
+            )
           )
         );
 
         const blockHeadersList =
           await opBlockHeaderService.$getBlockHeadersByHeights(
             'nbdr',
-            confirmedBlocks
+            Array.from(new Set(confirmedBlocks))

```

Comment:
why do you convert `ConfirmedBlockHeight[]` into `Set<ConfirmedBlockHeight>` and back to `ConfirmedBlockHeight[]`? to remove duplicates?

Reply:
Yes, correct